components/LogWorkout/LogWorkout.jsx

```diff
@@ -77,6 +81,56 @@ const LogWorkout = (props) => {
 
   const [exerciseState, setExerciseState] = useState(initialExerciseState);
 
+  // const currentUser = firebase.auth().currentUser.uid;
+  const currentUser = '68w6wWz8l5QJO3tDukh1fRXWYjD2';
+
+  const dbRef = firebase.firestore();
+  const userRef = dbRef.collection('users').doc(currentUser);
+
+  const incrementSelectedCycleIdx = () => {
+    // userRef.update({selectedCycleIndex: 2})
+    // userRef.get().then(function(doc) {
+    //       console.log("Document data:", doc.data().selectedCycleIndex);
+    //   } 
+    // )
+    return (
+    <CurrentCycle
+          rightPress={() => { dispatch({ type: INCREMENT_SELECTED_CYCLE_INDEX, cycleLength: cycleDetails.length }); }}
+        /> )
+  };
+
+  const sendWorkoutLogToDB = () => {
+
+    const workoutRecsRef = userRef.collection('workouts').doc(workoutId).collection('workoutRecords');
+  
+    //  newWorkoutLog = JSON.parse(JSON.stringify(newWorkoutLog, (k, v) => {
+    //   if (v === undefined) { return null; } return v;
+    // })); // This is needed so values can be undefined
+  
+    //make new object to only return data we need
+    const newWorkoutLog = {
+      date: firebase.firestore.FieldValue.serverTimestamp(),
+      exercises: exerciseState.map((exercise) => {
+          return {
+            exerciseId: exercise.ID,
+            sets: exercise.sets.map((set) => {
+                const parsedSet = {
+                  weight: set.weight || set.prevWeight,
+                }
+                _.set(parsedSet, exercise.type === 'REPS' ? ['reps'] : ['time'], set.duration || set.prevDuration)
+                return parsedSet;
+              })
+          }
+
+      })
+  
+    }
+    userRef.collection('workouts').doc(workoutId).update(newWorkoutLog);
```

Comment:
I don't think this currently updates lastPerformed, could you make sure that gets updated as well